### PR TITLE
DAOS-16078 common: fix a Coverity issue (release/2.6 catching up with the master)

### PR DESCRIPTION
**Pull request by janekmi**
### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by janekmi** [comment]
The source commit for this backport: https://github.com/daos-stack/daos/pull/14704

**Comment by janekmi** [comment]
Only known failures were observed:

- test_osa_offline_reintegration_without_checksum:DAOS-15608
- test_dfuse_daos_build_wb:DAOS-16215

**Review by wangshilong — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by janekmi** [comment]
> a backport should just do cherrypick of the master PR. it sounds like this is not a cherrypick and you have also added more changes than the original PR. please explain the extra change.

The extra commit is a revert of #14639 which happened to land on the release/2.6 branch but not on the master branch (I guess this is the root cause of the confusion). It did not land on the master branch as well since I realized it did not fix the actual issue.

So, in the sense a backport is an act of making two branches equal in some regard one might call it a backport. But I don't mind calling it differently.

**Comment by mchaarawi** [comment]
> The extra commit is a revert of #14639 which happened to land on the release/2.6 branch but not on the master branch (I guess this is the root cause of the confusion). It did not land on the master branch as well since I realized it did not fix the actual issue.
> 
> So, in the sense, that it makes two branches equal in some regard one might call it a backport. But I don't mind calling it differently.

let me rephrase. we cannot land anything to 2.6 that is not backported from master. i understand this PR contains a coverity fix and a revert of another. for the latter, where is the master PR for that? did that revert land on master?

**Comment by janekmi** [comment]
This PR comprises three commits:

1. https://github.com/daos-stack/daos/pull/14755/commits/4e7f49fcf343379d7cad28ed4071306b21160b36
2. https://github.com/daos-stack/daos/pull/14755/commits/650a9cf863bbac1ed190ddecfca20398ec23b44d
3. https://github.com/daos-stack/daos/pull/14755/commits/564f9d0f68a3f761e09883000aa95b0833606223

(1) is a revert of #14639 which has never landed on the master branch since it didn't work. So, reverting it is a step in the right direction, making release/2.6 and master equal and removing defunct code.

(2) and (3) are exactly what #14704 introduced to the master branch.

> i understand this PR contains a coverity fix and a revert of another. for the latter, where is the master PR for that? did that revert land on master?

Yes, (3) is a Coverity fix. (1) and (2) are reverts of failed attempts to fix the same Coverity issue.

Master PRs for the reverts:

- for (1) was unnecessary since #14639 never landed on the master branch.
- for (2) a master PR was #14704.

**Comment by mchaarawi** [comment]
> (1) is a revert of #14639 which has never landed on the master branch since it didn't work. So, reverting it is a step in the right direction, making release/2.6 and master equal and removing defunct code.

ok i get it now. yea the problem lies in this titem here. this should have never landed to 2.6 in the first place. the process is always to get a fix landed to master first then seek merge approval to land the backport to 2.6. so the process was broken for that PR. anyway, thank you for fixing that here.
